keps/0002-kong-gateway-api.md

```diff
@@ -171,41 +178,86 @@ spec:
+The above example is effectively the MVP for `Gateway` support, in that it would operationally function exactly like a default KIC deployment does now (the kong proxy is in the same pod as the controller and data-plane configurations occur over the same network namespace's via localhost) while also being explicit about the operational mode which will be documentative, help maintain a separate code path for this operational mode, and enable validation code in the early iterations to provide clear errors to the end-user.
 
 [chart]:https://github.com/kong/charts
 [kong-admin-api]:https://docs.konghq.com/gateway-oss/latest/admin-api/
-[gateway-spec-addrs]:https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/gateway_types.go#L428
-[gateway-spec-addr-types]:https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1alpha2/gateway_types.go#L452
+[anns]:https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+
+##### Operational Mode 2: Provisioned Gateways
+
+**NOTE**: required for milestone 3 in [graduation criteria](#graduation-criteria)
+
+TODO: work on this operational mode has not been started yet. At the time of writing our [operator][kong-operator] was still unready to take on this functionality, and maintainers were not comfortable with adding this operator-style functionality to the KIC.
 
-##### Operational Mode 2: Pre-Provisioned Gateways
+[kong-operator]:https://github.com/kong/kong-operator
 
-WIP: we're looking at support for gateway lifecycle management for the long term and this is here for posterity so readers know we're thinking about it, but in the short-term and for our initial implementation we're focused on `Operational Mode 1: In-Cluster Gateways` first and we'll come back to this mode in later iterations.
+###### Metrics
+
+**NOTE**: required for milestone 3 in [graduation criteria](#graduation-criteria)
+
+TODO
+
+###### Automatic Upgrades, HealthChecking and Rollbacks
+
+**NOTE**: required for milestone 3 in [graduation criteria](#graduation-criteria)
+
+TODO
 
 ##### Additional Considerations
 
 - https://github.com/Kong/kubernetes-ingress-controller/issues/702 is related to our single controller multi-proxy lifecycle management concerns
 
 #### HTTPRoute Controller
 
-The HTTPRoute controller is fairly straightforward, which backend proxy is responsible is based on which Gateway its attached to in the `parentRefs`, but then the resource otherwise
-gets parsed, converted into Kong types, and posted to `/config` like any other API resource that currently exists.
+**NOTE**: required for milestone 1 in [graduation criteria](#graduation-criteria)
+
+The HTTPRoute controller is fairly straightforward, which backend proxy is responsible is based on which Gateway its attached to in the `parentRefs`, but then the resource otherwise gets parsed, converted into Kong types, and posted to `/config` like any other API resource that currently exists.
```

Comment:
Is this (_and posted to `/config`_) to mean that we intend to drop DB mode for GW API?

Reply:
If you mean am I pushing for a direction where we use the existing configuration option, and NOT take the time to write the REST layer for DB mode and then take advantage of that as a part of this effort, the answer is yes. The long and short of it is that I feel we're quite behind the curve on implementing Gateway APIs and I want to push for a very lightweight, light touch milestone 1.

Reply:
```suggestion
The HTTPRoute controller is fairly straightforward, which backend proxy is responsible is based on which Gateway its attached to in the `parentRefs`, but then the resource otherwise gets parsed (by `parser.Build()`), converted into Kong types, and posted to Kong like any other API resource that currently exists, using the `KongState`, proxy and `sendconfig` abstractions as they are used today.
```